dataclass_rest/parse_func.py

```diff
@@ -1,16 +1,21 @@
 import string
 from inspect import getfullargspec, FullArgSpec, isclass
-from typing import Callable, List, Sequence, Any, Type, TypedDict, Dict
+from typing import Callable, List, Sequence, Any, Type, TypedDict, Dict, Union, Optional
 
 from .http_request import File
 from .methodspec import MethodSpec
 
 DEFAULT_BODY_PARAM = "body"
 
 
-def get_url_params(url_template: str) -> List[str]:
-    parsed_format = string.Formatter().parse(url_template)
-    return [x[1] for x in parsed_format]
+def get_url_params(url_template: Union[str, Callable[..., str]], arg_spec: Optional[FullArgSpec] = None) -> List[str]:
```

Comment:
I'd rather split into 2 functions. It's strange to get `arg_spec` for string.
Additionally we can remove try/except below and just call 2 different functions for both cases

Reply:
done